### PR TITLE
Support aarch64 docker job and general architecture support

### PR DESCRIPTION
**Pull request by jdsn**
untested so far
just waiting for feedback about the change

### CODE REVIEW

**Comment by bmwiedemann** [comment]
+1 Looks mostly nice. At some point, we might want to use openstack-mkcloud-SLE12-{arch} also for x86_64.
And I guess it will need some testing.

**Comment by jdsn** [comment]
> At some point, we might want to use openstack-mkcloud-SLE12-{arch} also for x86_64.
> And I guess it will need some testing.

I am fine with changing it to openstack-mkcloud-SLE12-x86_64 with this PR as well. This makes the arch explicit (so no defauting to x86_64).

@rsalevsky @dirkmueller are you fine with this as well?

**Comment by rsalevsky** [comment]
@jdsn I'm fine with this.

File: scripts/jenkins/jobs-ibs/cloud-mkcloud7.yaml [hunk 1]
```diff
@@ -2,18 +2,28 @@
...
+        - 'cloud-mkcloud{version}-job-{arch}-ha'
+        - 'cloud-mkcloud{version}-job-{arch}-hyperv'
+        - 'cloud-mkcloud{version}-job-{arch}-raid'
+        - 'cloud-mkcloud{version}-job-{arch}-ssl'
+        - 'cloud-mkcloud{version}-job-{arch}-tempestfull'
+        - 'cloud-mkcloud{version}-job-{arch}-upgrade'
+        - 'cloud-mkcloud{version}-job-{arch}-sle12sp2'
+        - 'cloud-mkcloud{version}-job-{arch}-xen'
+- project:
+    name: cloud-mkcloud7
```
Comment (dirkmueller) [inline scripts/jenkins/jobs-ibs/cloud-mkcloud7.yaml]: whats this supposed to be ? missing -aarch64 suffix perhaps?

**Comment by dirkmueller** [comment]
imho this breaks cloud 6 in the way its done right now.. -1

File: scripts/jenkins/jobs-ibs/templates/cloud-mkcloud-gating-template.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 - job-template:
-    name: 'cloud-mkcloud{version}-gating'
+    name: 'cloud-mkcloud{version}-{arch}-gating'
```
Comment (dirkmueller) [inline scripts/jenkins/jobs-ibs/templates/cloud-mkcloud-gating-template.yaml]: hmm, why do we want to do gating based on architecture? the sources are shared across architectures

Comment (jdsn) [inline scripts/jenkins/jobs-ibs/templates/cloud-mkcloud-gating-template.yaml]: We are not. This is just a rename - to have a consistent naming scheme. And it makes it explicit what we are testing.

And yes, it (the rename) allows to have gating run on other architectures in the future as well.

File: scripts/jenkins/jobs-ibs/cloud-mkcloud7.yaml (hunk 1 above)
Comment (jdsn) [inline scripts/jenkins/jobs-ibs/cloud-mkcloud7.yaml]: See 3 lines below. There `arch` is defined.

**Comment by jdsn** [comment]
> imho this breaks cloud 6 in the way its done right now.. -1

@dirkmueller how/where does it break cloud 6?

**Comment by jdsn** [comment]
rebased on latest master

File: scripts/jenkins/jobs-ibs/templates/cloud-mkcloud-gating-template.yaml (hunk 2 above)
Comment (dirkmueller) [inline scripts/jenkins/jobs-ibs/templates/cloud-mkcloud-gating-template.yaml]: Well, that's my point - I think we should have only one gating job that at some point should also run tests on non-x86. If you do a gating per architecture then you have a pass on x86 causing a submit even though aarch64 is broken.

**Comment by dirkmueller** [comment]
@jdsn: you're renaming all template defined jobs which are shared between cloud 6 and 7, but you only adjust cloud7.

**Comment by dirkmueller** [comment]
also, I wonder if -{arch} shouldn't be a suffix rather than an infix. this way the related jobs are listed together.

**Comment by jdsn** [comment]
> also, I wonder if -{arch} shouldn't be a suffix rather than an infix. this way the related jobs are listed together.

Well its just a different relation :) With the infix there still is a relation between the jobs (the architecture).

But I don't care. Any objections to change arch to a suffix: @bmwiedemann @rsalevsky ?

**Comment by rsalevsky** [comment]
@jdsn I'm fine.

**Comment by jdsn** [comment]
changed arch string to suffix, rebased on latest master, renamed also Cloud6 jobs to not break the templates

**Comment by dirkmueller** [comment]
@jdsn could you please rebase?

**Comment by dirkmueller** [comment]
@toabctl this is a bit of an intrusive change (we need to manually go in and rename a ton of jobs before deploying this), let me know when would be a convenient time for this.

**Comment by jdsn** [comment]
> @jdsn could you please rebase?

yes, will do

**Comment by dirkmueller** [comment]
+1

**Comment by dirkmueller** [comment]
(imho 1st and 3rd commit could be squashed together)

**Comment by bmwiedemann** [comment]
+0.5 looks OK to me, but I dont have much jjb-experience.

**Comment by toabctl** [comment]
+1

**Comment by dirkmueller** [comment]
deployed, looks good so far

**Comment by dirkmueller** [comment]
closed in favor of https://github.com/SUSE-Cloud/automation/pull/1075

**Pull request closed without merging**